### PR TITLE
Changes to build and initialization settings to get ubuffer working with Halide-to-Hardware on travis

### PR DESCRIPTION
**Pull request by dillonhuff**
@joyliu37 @jeffsetter I now have the unified buffer branch building in halide to hardware. Some changes were needed to please gccs warning settings and get the makefile to work on travis.

### CODE REVIEW

File: cfunc/Makefile [hunk 1]
```diff
@@ -1,8 +1,11 @@
-CC = g++
+
+CC := $(if $(CXX),$(CXX), g++)
 
 INCLUDE_DIR = include
 BUILD_DIR = bin
-COREIR_DIR = /nobackup/joeyliu/aha/coreir
+COREIR_DIR := $(if $(COREIR_DIR),$(COREIR_DIR),/nobackup/joeyliu/aha/coreir)
```
Comment (jeffsetter) [inline cfunc/Makefile]: Can we changes this to 
COREIR_DIR ?= ../../coreir

Comment (dillonhuff) [reply cfunc/Makefile]: @jeffsetter just changed it.

Comment (jeffsetter) [reply cfunc/Makefile]: Can we change it to ../../coreir
instead of Joey's coreir folder?

Comment (joyliu37) [reply cfunc/Makefile]: So do I need to change to ../../coreir to pass the travis?

Comment (jeffsetter) [reply cfunc/Makefile]: Not for travis (since it will likely defined COREIR_DIR), but it would be best to change it for anyone that clones this repo.

**Pull request merged**